src/trans.rs

```diff
@@ -549,6 +720,74 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
         }
     }
 
+    fn generate_runtime_start(&mut self, entry_fn: &str) -> BinaryenFunctionRef {
```

Comment:
Would it be possible to move this into a library that is always (or at least, by default) linked in? It seems like it might be similar to [rtstartup](https://github.com/rust-lang/rust/tree/master/src/rtstartup). I guess the answer to this probably ultimately comes down to what do we do about libcore, libstd, etc.